crates/extensions/tedge_p11_server/src/main.rs

```diff
@@ -0,0 +1,163 @@
+//! thin-edge.io PKCS#11 server.
+//!
+//! The purpose of this crate is to allow thin-edge services that possibly run in containers to access PKCS#11 tokens in
+//! all of our supported architectures.
+//!
+//! There are 2 main problems with using a PKCS#11 module directly by thin-edge:
+//! 1. One needs to use a dynamic loader to load the PKCS#11 module, which is not possible in statically compiled musl
+//! 2. When thin-edge runs in a container, additional setup needs to be done by the user to expose cryptographic tokens
+//!    in the container, using software like p11-kit.
+//!
+//! To avoid extra dependencies and possibly implement new features in the future, it was decided that thin-edge.io will
+//! provide its own bundled p11-kit-like service.
+
+use std::io::{BufRead, BufReader, Read, Write};
+use std::os::unix::net::{UnixListener, UnixStream};
+use std::path::Path;
+use std::sync::Arc;
+
+use anyhow::Context;
+use camino::Utf8PathBuf;
+use certificate::parse_root_certificate::pkcs11::PkcsSigner;
+use certificate::parse_root_certificate::{pkcs11, CryptokiConfigDirect};
+use clap::command;
+use clap::Parser;
+use tracing::debug;
+use tracing::info;
+use tracing::level_filters::LevelFilter;
+use tracing_subscriber::EnvFilter;
+
+/// thin-edge.io service for passing PKCS#11 cryptographic tokens.
+#[derive(Debug, Clone, PartialEq, Eq, Parser)]
+#[command(version)]
+pub struct Args {
+    /// A path where the service's unix token will be created.
+    #[arg(default_value = "./thin-edge-pkcs11.sock")]
```

Comment:
Default to the same name as the service as it makes it easier for users to reason with (what thing created the socket) (and hard coding is fine here).

```suggestion
    #[arg(default_value = "./tedge-p11-server.sock")]
```